From d5ac132e68626ba76fdb6387b5b6ad3b54e0cef2 Mon Sep 17 00:00:00 2001
From: Lawrence Rust <lvr@softsystem.co.uk>
Date: Thu, 17 Mar 2016 13:53:10 +0000
Subject: [PATCH 314/348] MythCoreContext::IsConnectedToMaster test server
 socket connection state

This patch adds a test on the server socket connection status.
Currently the function returns true when the socket is disconnected.

This primarily affects mythwelcome and mythlcdserver.

Signed-off-by: Lawrence Rust <lvr@softsystem.co.uk>
---
 mythtv/libs/libmythbase/mythcorecontext.cpp |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mythtv/libs/libmythbase/mythcorecontext.cpp b/mythtv/libs/libmythbase/mythcorecontext.cpp
index 090c95c..7442a71 100644
--- a/mythtv/libs/libmythbase/mythcorecontext.cpp
+++ b/mythtv/libs/libmythbase/mythcorecontext.cpp
@@ -578,7 +578,7 @@ MythSocket *MythCoreContext::ConnectEventSocket(const QString &hostname,
 bool MythCoreContext::IsConnectedToMaster(void)
 {
     QMutexLocker locker(&d->m_sockLock);
-    return d->m_serverSock;
+    return d->m_serverSock && d->m_serverSock->IsConnected();
 }
 
 void MythCoreContext::BlockShutdown(void)
-- 
1.7.9.5

